src/components/sidenav/sidenav.spec.js

```diff
@@ -173,6 +173,85 @@ describe('mdSidenav', function() {
 
   });
 
+  describe("focus", function() {
+
+    var $material, $mdInteraction;
+
+    beforeEach( inject(function(_$material_, _$mdInteraction_) {
+      $material = _$material_;
+      $mdInteraction = _$mdInteraction_;
+    }));
+
+    function flush() {
+      $material.flushInterimElement();
+    }
+
+    function setupTrigger() {
+      var el;
+      inject(function($compile, $rootScope) {
+        var parent = angular.element(document.body);
+        el = angular.element('<button aria-label="Toggle Menu">Toggle</button>');
+        parent.append(el);
+        $compile(parent)($rootScope);
+        $rootScope.$apply();
+      });
+      return el;
+    }
+
+    it("should restore after sidenav triggered by keyboard", function(done) {
+      var sidenavElement = setup('');
+      var triggerElement = setupTrigger();
+      var controller = sidenavElement.controller('mdSidenav');
+
+      triggerElement.focus();
+
+      var evt = document.createEvent("KeyboardEvent");
+      evt.initEvent("keydown", true, true, window, 0, 0, 0, 0, 13, 13);
+      triggerElement[0].dispatchEvent(evt);
+
+      controller.$toggleOpen(true);
+      flush();
+
+      triggerElement.blur();
+
+      expect(document.activeElement).not.toBe(triggerElement[0]);
+
+      controller.$toggleOpen(false);
+      flush();
+
+      expect($mdInteraction.getLastInteractionType()).toBe("keyboard");
+      expect(document.activeElement).toBe(triggerElement[0]);
+      done();
+    });
+
+    it("should not restore after sidenav triggered by mouse", function(done) {
+      var sidenavElement = setup('');
+      var triggerElement = setupTrigger();
+      var controller = sidenavElement.controller('mdSidenav');
+
+      triggerElement.focus();
+
+      var event = document.createEvent("MouseEvent");
+      event.initMouseEvent("mousedown", true, true, window, null, 0, 0, 0, 0, false, false, false, false, 0, null);
```

Comment:
Yes it was difficult to get this runnning on Phantom.js. It took a lot of time in getting the test working. I already tried that ````triggerHandler``` but then the interaction service, won't get triggered because its not notifying the parent nodes. (this imitates a ```real``` click).